server/src/QRServer/lobby/lobbyclient.py

```diff
@@ -110,11 +110,16 @@ async def _handle_server_recent(self, _: ServerRecentRequest):
         await self.send_msg(self.lobby_server.get_last_logged())
         await self.send_msg(LastPlayedResponse.new(recent_matches))
 
-    async def _handle_server_ranking(self, _: ServerRankingRequest):
-        await self.send_msg(ServerRankingThisMonthResponse.new([
-            RankingEntry(player='test', wins=12, games=30),
-            RankingEntry(player='test2', wins=2, games=2),
-        ]))
+    async def _handle_server_ranking(self, request: ServerRankingRequest):
+        start_date, end_date = utils.make_month_dates(request.get_month(), request.get_year())
+
+        rankings = self.connector.get_ranking(
+            start_date=start_date,
+            end_date=end_date,
+            ranked_only=self.config.leaderboards_ranked_only.get(),
+            include_void=self.config.leaderboards_include_void.get()
+        )
+        await self.send_msg(ServerRankingThisMonthResponse.new(rankings))
```

Comment:
Should we always return `ServerRankingThisMonthResponse`? There is a dedicated structure for non-this-month response

Reply:
#tocheck

Reply:
I did some digging and the way it works right now seems to be okay, regardless of which month is selected. I couldn't find the other structure, but it seems to be unnecessary.